### PR TITLE
[3.0][cdc-common] Introduce Event DataSink APIs for Flink CDC 3.0

### LINKED ISSUE #2626: [flink-cdc-common] Introduce DataSink related APIs
### Search before asking

- [X] I searched in the [issues](https://github.com/ververica/flink-cdc-connectors/issues) and found nothing similar.

### Motivation

Introduce DataSink related APIs

### Solution

_No response_

### Alternatives

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by Jiabao-Sun**
Introduce Event DataSink APIs for Flink CDC 3.0

This closes #2626

### CODE REVIEW

**Comment by Jiabao-Sun** [comment]
@lvyanquan PTAL.

**Review by lvyanquan — APPROVED** [review]

File: flink-cdc-common/src/main/java/com/ververica/cdc/common/sink/DataSink.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.common.sink;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/**
+ * {@code DataSink} is used to write change data and apply metadata changes by {@link
+ * MetadataApplier} to external systems.
```
Comment (leonardBang) [inline flink-cdc-common/src/main/java/com/ververica/cdc/common/sink/DataSink.java:23]:
```suggestion
 * {@code DataSink} is used to write change data to external system and apply metadata changes to external systems as well.
```

File: flink-cdc-common/src/main/java/com/ververica/cdc/common/sink/DataSink.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+@PublicEvolving
+public interface DataSink {
+    EventSinkProvider getEventSinkProvider();
```
Comment (leonardBang) [inline flink-cdc-common/src/main/java/com/ververica/cdc/common/sink/DataSink.java:27]: A blank line and method java doc are welcome

**Review by leonardBang — COMMENTED** [review]
Thanks @Jiabao-Sun for the contribution, I only left two comments

**Review by leonardBang — APPROVED** [review]
Thanks @Jiabao-Sun for the update, LGTM

**Pull request merged**